packages/frontend/src/context/User.ts

```diff
@@ -106,6 +110,59 @@ export class User {
         return this.currentEpoch
     }
 
+    async loadRecords() {
+        const epksBase10 = this.allEpks.map((epk) =>
+            BigInt('0x' + epk).toString()
+        )
+
+        const apiURL = makeURL(`records/${epksBase10.join('_')}`, {})
+        const r = await fetch(apiURL)
+        const data = await r.json()
+        const rawRecords = data.map((r: Record) => {
+            return {
+                ...r,
+                from: BigInt(r.from)
+                    .toString(16)
+                    .padStart(this.unirepConfig.epochTreeDepth / 4, '0'),
+                to: BigInt(r.to)
+                    .toString(16)
+                    .padStart(this.unirepConfig.epochTreeDepth / 4, '0'),
+            }
+        })
+
+        this.recordsByEpk = {} // maybe it's not a good solution
```

Comment:
Hard to follow what this code is doing considering there is already a state variable with the name `recordsByEpk`

Reply:
since it will reload the records from db each time this function is called, and if it wasn't cleared, the records are duplicated. ofc we can do things like if a record exists already then just don't do anything to it, but it costs time to do this every time, so i choose to just let it be empty again and add records into it.